tests/Jobs/PtpJobTest.php

```diff
@@ -21,94 +21,182 @@ class PtpJobTest extends TestCase
-        $this->inputFile = 'test';
-        $inputFileContent = [
+
+        //Add an annotation that is not a point annotation to check that it is filtered out
+        $this->fakeAnnotation = ImageAnnotation::factory()->create([
+            'image_id' => $this->image->id,
+            'shape_id' => Shape::polygonId(),
+            'points' => [0,0,1,2,3,4,5],
+        ]);
+
+        $fakeLabel = Label::factory()->create();
+
+        ImageAnnotationLabel::factory()->create([
+            'annotation_id' => $this->fakeAnnotation->id,
+            'label_id' => $fakeLabel->id,
+            'user_id' => $this->user2->id,
+        ]);
+
+        $this->inputFileContents = [
             $this->image->id => [[
                 'annotation_id' => $this->imageAnnotation->id,
-                'points' => [1,2],
+                'points' => $this->imageAnnotation->points,
                 'shape' => Shape::pointId(),
                 'image' => $this->image->id,
                 'label' => $this->label->id,
-            ]]];
-        $this->outputFile = 'testOutput.json';
-        $outputFileContent = [[
-            'points' => [1,2,3,4,5,6],
-            'annotation_id' => $this->imageAnnotation->id,
-            'label_id' => $this->label->id,
-        ]];
-
-        config(['ptp.ptp_storage_disk' => 'test']);
-        config(['ptp.device' => 'cpu']);
-        $storage = Storage::disk('test');
-        $storage->put(config('ptp.temp_dir').'/'.$this->inputFile.'.json', json_encode($inputFileContent));
-        file_put_contents(config('ptp.temp_dir').'/'.$this->outputFile, json_encode($outputFileContent));
+        ]]];
     }
 
     public function testPtpHandle(): void
     {
         //Test that the PTP job handle correctly calls the handle, succedes and cleans up the volume
-        $job = new MockPtpJob($this->inputFile, $this->outputFile, $this->user, $this->uuid);
-        $job->handle();
-        $this->assertTrue($job->pythonCalled);
-        $volume = Volume::where('id', $this->volume->id)->first();
-        $this->assertFalse(isset($volume->attrs['ptp_job_id']));
+        $job = new MockPtpJob(
+            $this->volume->id, $this->volume->name, $this->inputFile, $this->outputFile, $this->user, $this->uuid
+        );
+
+        $this->expectException(Exception::class);
+        $this->expectExceptionMessage('No files converted!');
+        try {
+            file_put_contents(config('ptp.temp_dir').'/'.$this->outputFile, '[]');
+            $this->setUpAnnotations();
+            $job->handle();
+            $this->assertTrue($job->pythonCalled);
+            $volume = Volume::where('id', $this->volume->id)->first();
+            $this->assertFalse(isset($volume->attrs['ptp_job_id']));
+        } catch(Exception $e) {
+            throw ($e);
+        } finally {
+            unlink(config('ptp.temp_dir').'/'.$this->inputFile.'.json');
+            unlink(config('ptp.temp_dir').'/'.$this->inputFile.'_images.json');
+        }
```

Comment:
I said above that you don't need the input/output paths as constructor parameter any more but here you use them for testing. What you con do instead is to update the mock class below so you can set something like a public `inputFileContent` array. Also you can update the mocked `python()` method to write something to a public `outputFileContent` variable in the mock class.

This way you avoid test code in the actual implementation.